x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/Monitoring.java

```diff
@@ -77,6 +77,9 @@
     public static final Setting<Boolean> CLEAN_WATCHER_HISTORY = boolSetting("xpack.watcher.history.cleaner_service.enabled",
         true, Setting.Property.Dynamic, Setting.Property.NodeScope, Setting.Property.Deprecated);
 
+    public static final Setting<Boolean> MIGRATION_DECOMMISSION_ALERTS = boolSetting("xpack.monitoring.migration.decommission_alerts",
```

Comment:
It feels kinda weird to have this AND `xpack.monitoring.exporters.*.cluster_alerts.management.enabled` setting. 

I wonder if makes sense to 
1) expose a POST _monitoring/cluster_alert/disable endpoint (or whatever the name is)
2) when that is called iterate through all of the current exporters and set `xpack.monitoring.exporters.<i>.cluster_alerts.management.enabled=false` and set some cluster state to no longer try to install cluster alerts (and reach out to delete the cluster alerts for those exporters)
3) If a new exporter is created, set `xpack.monitoring.exporters.<new_exporter>.cluster_alerts.management.enabled=false` based on the cluster state. 
4) expose a POST _monitoring/cluster_alert/enable

thoughts ?

Reply:
I'd be in favour of less settings but OTOH current flow with this setting seems simpler and less error-prone. And currently you can't enable single exporter back (with `xpack.monitoring.exporters.<i>.cluster_alerts.management.enabled=true`) which I think is good thing (easier to reason about cluster state).
We can still expose `_monitoring/cluster_alert/disable` endpoint if needed though if we want, it may be easier for the user to just call API than looking up setting to use

Reply:
I consider this more of a first step with a new action that performs the migration to follow. If the exporters are disabled, this setting only ensures that the alerts are not created if they are turned back on (since alerts are created lazily much to my chagrin).

The main concern that was raised in the migration issue was that if a new exporter is configured, either on purpose or by accident, the cluster alerts for that exporter will not be installed. A global setting to control the existence of the alerts is the easiest way to fix that, even if it overlaps with other settings.

Reply:
that makes sense, thanks for explaining.